### PR TITLE
Fixing up IRGenModule to handle changes in CodeGenOptions.

### PR DESCRIPTION
**Pull request by plotfi**
DisableFPElim was removed from CGO in https://reviews.llvm.org/D56353
This change fixes the build for master-next.

### CODE REVIEW

File: lib/IRGen/IRGenModule.cpp [hunk 1]
```diff
@@ -94,7 +94,14 @@ static clang::CodeGenerator *createClangCodeGenerator(ASTContext &Context,
 
   auto &CGO = Importer->getClangCodeGenOpts();
   CGO.OptimizationLevel = Opts.shouldOptimize() ? 3 : 0;
-  CGO.DisableFPElim = Opts.DisableFPElim;
+  clang::CodeGenOptions::FramePointerKind FP;
+  if (!Opts.DisableFPElim) {
+    FP = clang::CodeGenOptions::FramePointerKind::None;
+  } else {
+    FP = clang::CodeGenOptions::FramePointerKind::All;
+  }
```
Comment (compnerd) [inline lib/IRGen/IRGenModule.cpp]: Any reason to not make this a ternary?

```
CGO.setFramePointer(Opts.DisableFPElim
                        ? clang::CodeGenOptions::FramePointerKind::All
                        : clang::CodeGenOptions::FramePointerKind::None);
```

File: lib/IRGen/IRGenModule.cpp [hunk 2]
```diff
@@ -895,12 +902,13 @@ bool swift::irgen::shouldRemoveTargetFeature(StringRef feature) {
...
+  StringRef FpKind;
   if (!IRGen.Opts.DisableFPElim) {
-    Attrs.addAttribute("no-frame-pointer-elim", "false");
+    FpKind = "none";
   } else {
-    Attrs.addAttribute("no-frame-pointer-elim", "true");
-    Attrs.addAttribute("no-frame-pointer-elim-non-leaf");
+    FpKind = "all";
   }
+  Attrs.addAttribute("frame-pointer", FpKind);
```
Comment (compnerd) [inline lib/IRGen/IRGenModule.cpp]: Similar.

```
Attrs.addAttribute("frame-pointer", IRGen.Opts.DisableFPElim ? "all" : "none");
```

**Comment by compnerd** [comment]
Also, please don't reference the differential, but rather than the SVN revision in the commit message.

File: lib/IRGen/IRGenModule.cpp (hunk 1 above)
Comment (plotfi) [reply lib/IRGen/IRGenModule.cpp]: No, was just  trying to keep the structure of the code as similar to how it was before.

**Review by compnerd — APPROVED** [review]
Would be nice to clang-format this, but LGTM.

**Pull request merged**